### PR TITLE
detect all available devices for bluestore by default

### PR DESCRIPTION
**Pull request by travisn**
If `ROOKD_DATA_DEVICES=all`, configure all devices for bluestore that meet the following criteria:
- Do not have a file system (such as a system partition
- Is of type Disk
- Has no parent device

If no devices are found, a filestore directory will be created

### CODE REVIEW

**Comment by jbw976** [comment]
Needs rebasing from upstream master now

File: pkg/cephmgr/osd/agent.go [hunk 1]
```diff
@@ -70,10 +72,22 @@ func (a *osdAgent) Name() string {
 // set the desired state in etcd
 func (a *osdAgent) Initialize(context *clusterd.Context) error {
 
+	deviceCount := 0
 	if len(a.devices) > 0 {
-		// add the devices to desired state
-		devices := strings.Split(a.devices, ",")
+		var devices []string
+		if strings.EqualFold(a.devices, allDevices) {
```
Comment (jbw976) [inline pkg/cephmgr/osd/agent.go]: i hadn't come across EqualFold yet.  is that idiomatic golang for doing case insensitive comparisons?

**Comment by bassam** [comment]
How do we ensure that we don't incorrectly wipe devices?

**Comment by travisn** [comment]
@bassam, if you use --data_devices=all, we will format all the devices that don't have a file system. If you don't want it to take over all your devices, you would have to name the devices with sda,sdb,... Another check we could add is to skip devices that are already partitioned.

File: pkg/cephmgr/osd/agent.go (hunk 1 above)
Comment (travisn) [inline pkg/cephmgr/osd/agent.go]: correct, case insensitive

**Comment by bassam** [comment]
@travis how do we check that the device has no file system? If it's used as a raw device that would be hard to find out.

**Comment by travisn** [comment]
right, we wouldn't be able to check on raw devices. Perhaps this is where we should have a flag --are-you-really-sure

**Comment by bassam** [comment]
yes adding some flag would be helpful. can we even tell if the device is already running bluestore, or are we relying on the existence of a partition table.

**Comment by travisn** [comment]
to tell if it is already running bluestore we rely on the osd config folder under the data dir

**Comment by bassam** [comment]
I think we should hold off on this one until we have a stricter check of whether a disk is really raw, or until there's a way for the user to tell us that they really OK with us destroying data.

**Comment by travisn** [comment]
a flag is added to confirm if they really want to format all the devices

**Comment by travisn** [comment]
This is the change for https://github.com/rook/rook/issues/103

**Comment by travisn** [comment]
For design of the change, comments added to https://github.com/rook/rook/issues/103

**Comment by bassam** [comment]
so with this change the incremental improvement is the ability to specify "all" vs. "sda,sdb,sdc", right?

If so, then I would recommend that we should hold this change until we have a more complete design as you outlined in #103. Not sure this change is useful on its own.

**Comment by travisn** [comment]
Besides making the test scenario simpler, this does have the nice side effect of skipping the system volume. Right now you don't know which volume is the system so you just have to give vagrant all the devices (sda,sdb,sdc,sdd). Rook will just fail to format the system volume in that case and it shows up as an osd that is down. With the change we will have all osds that we expect.

**Comment by bassam** [comment]
IMO this is a small improvement. I'd rather us focus on getting something more meaningful down the road.

**Comment by travisn** [comment]
Another thought is to disable the `all` option, but leave the other changes. Then we still get the benefits of skipping the system volume and also the foundation for other changes such as https://github.com/rook/rook/issues/305. Then when we get to https://github.com/rook/rook/issues/103 we can reconsider the `all` option.

**Comment by bassam** [comment]
sounds reasonable.

**Comment by travisn** [comment]
Relevant functionality that is still needed is part of https://github.com/rook/rook/pull/318, but without the devices=`all` option.

**Pull request closed without merging**